tools/depmod.c

```diff
@@ -1308,14 +1306,13 @@ static int depmod_modules_search_dir(struct depmod *depmod, DIR *d, size_t basel
 			continue;
 
 		namelen = strlen(name);
-		if (scratchbuf_alloc(s_path, baselen + namelen + 2) < 0) {
+		if (!strbuf_reserve(path, namelen + 2)) {
 			err = -ENOMEM;
 			ERR("No memory\n");
 			continue;
 		}
 
-		path = scratchbuf_str(s_path);
-		memcpy(path + baselen, name, namelen + 1);
+		strbuf_pushmem(path, name, namelen);
```

Comment:
Since `namelen = strlen(name)`, this can become `strbuf_pushchars`.

Reply:
Adding this note as a second comment to the same line: The API of strbuf as it was to be used before, would expect a check to see if adding something to strbuf was successful or not. Even if we could make a static analyzer smart enough to see that it fits ... Is it our job as a strbuf user to care about its internal status? A simple approach would be to wrap all strbuf functions with a check and we are done. Everything else expects us to be smart ... And at worst smarter than we really are. :)

Reply:
In fact, let me do it in an update so the compiler can see the 2 calls to strlen()

Reply:
Right, if we have needed data stored already, we can just reuse ist.